From 304a22e1e43f32cda33e60586b63d5c3fd1936d4 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Tue, 12 Sep 2017 17:28:48 +0800
Subject: [PATCH 4512/5242] MLK-19413-1 dt-bindings: display: imx-drm: Add
 pixel combiner descriptions

commit  33f8e7f13fed26cb15bbe431e5c96a0c82fcc3f2 from
https://source.codeaurora.org/external/imx/linux-imx.git

Pixel combiner found in i.MX8 SoCs may combine two display
streams(one master and the other slave) to drive a high
pixel rate display.  This patch adds DT property descriptions
in imx-drm DT documentation for pixel combiner.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../bindings/display/imx/fsl-imx-drm.txt           |   18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/Documentation/devicetree/bindings/display/imx/fsl-imx-drm.txt b/Documentation/devicetree/bindings/display/imx/fsl-imx-drm.txt
index 5ce9d59..bcd713e 100644
--- a/Documentation/devicetree/bindings/display/imx/fsl-imx-drm.txt
+++ b/Documentation/devicetree/bindings/display/imx/fsl-imx-drm.txt
@@ -132,6 +132,7 @@ Required properties:
 - power-domains: phandle pointing to power domain.
 - fsl,dpr-channels: phandles to the DPR channels attached to this DPU,
   sorted by memory map addresses.  Only valid for i.MX8qm and i.MX8qxp.
+- fsl,pixel-combiner: phandle to the pixel combiner unit attached to this DPU.
 Optional properties:
 - port@[0-1]: Port nodes with endpoint definitions as defined in
   Documentation/devicetree/bindings/media/video-interfaces.txt.
@@ -171,6 +172,7 @@ dpu: dpu@56180000 {
 	fsl,dpr-channels = <&dpr1_channel1>, <&dpr1_channel2>,
 			   <&dpr1_channel3>, <&dpr2_channel1>,
 			   <&dpr2_channel2>, <&dpr2_channel3>;
+	fsl,pixel-combiner = <&pixel_combiner1>;
 
 	dpu1_disp1: port@1 {
 		reg = <1>;
@@ -301,6 +303,22 @@ example:
 		};
 	};
 
+Freescale i.MX8 PC (Pixel Combiner)
+=============================================
+Required properties:
+- compatible: should be "fsl,<chip>-pixel-combiner"
+- reg: should be register base and length as documented in the
+  datasheet
+- power-domains: phandle pointing to power domain
+
+example:
+
+pixel-combiner@56020000 {
+	compatible = "fsl,imx8qm-pixel-combiner";
+	reg = <0x0 0x56020000 0x0 0x10000>;
+	power-domains = <&pd_dc0>;
+};
+
 Freescale i.MX8 PRG (Prefetch Resolve Gasket)
 =============================================
 Required properties:
-- 
1.7.9.5

